hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -1308,6 +1311,10 @@ object PruneDeadFields {
         val value2 = rebuildIR(value, BindingEnv.empty, memo)
         memo.relationalRefs += name -> value2.typ
         RelationalLetTable(name, value2, rebuild(body, memo))
+      case BlockMatrixToTableApply(bmir, aux, function) =>
+        val bmir2 = rebuild(bmir, memo)
+        val aux2 = rebuildIR(aux, BindingEnv.empty, memo)
+        BlockMatrixToTableApply(bmir2, aux2, function)
```

Comment:
I don't understand how PruneDeadFields works, but I'm surprised this is such a rote addition. Do we not have to note the new fields added by the function? e.g. `ibs1`

Reply:
Maybe @tpoterba should take a look at this too. Maybe I should be removed from the spin wheel for this stuff.

Reply:
Prune has two phases: analysis and rebuilding. 

In the analysis pass, we associate with each IR node of type `T` the smallest possible type `Tmin` it could produce to guarantee a valid computation.

In the rebuild pass, we do a bottom-up rebuild where each IR node is rebuilt with a type `Treb` such that `Tmin <= TTreb <= T`. There are cases where a node may not end up with the minimal possible type, generally when it would take computational work to upcast a node to that type. This pass carries around an environment of bindings, which contain the rebuilt types of defined variables.

The code here is totally correct. We view a `BlockMatrixToTableApply` node as one that demands the full type `T` from its children, and which will always return the full type `T` of the block matrix apply node to its parents.